tap_github.py

```diff
@@ -83,19 +87,34 @@ def get_bookmark(state, repo, stream_name, bookmark_key):
     return None
 
 def authed_get(source, url, headers={}):
-    with metrics.http_request_timer(source) as timer:
-        session.headers.update(headers)
-        resp = session.request(method='get', url=url)
-        if resp.status_code == 401:
-            raise AuthException(resp.text)
-        if resp.status_code == 403:
-            raise AuthException(resp.text)
-        if resp.status_code == 404:
-            raise NotFoundException(resp.text)
+    for _ in range(0, 3):  # 3 attempts
```

Comment:
Couldn't this be replaced with the `backoff` library?

Reply:
@mredolatti

Reply:
Hi @luandy64  i'm not familiar with that library, but i'll look into it as soon as possible and tidy up the PR.

Thanks and sorry for the delay!

Reply:
@mredolatti I sumbled on this example: https://github.com/singer-io/tap-exchangeratesapi/blob/master/tap_exchangeratesapi/__init__.py#L38

Reply:
I'll take a look at this tonight and see if I can put the branch up to speed

Reply:
@luandy64 @osterman could it be that the backoff library doesn't provide an easy way to use headers like: `X-RateLimit-Remaining, X-RateLimit-Reset` to determine how long to back off: https://github.com/litl/backoff/issues/38?